packages/react-components/components/NumberKeypad.tsx

```diff
@@ -0,0 +1,86 @@
+import Touchable from '@celo/react-components/components/Touchable'
+import Backspace from '@celo/react-components/icons/Backspace'
+import { fontStyles } from '@celo/react-components/styles/fonts'
+import * as React from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+
+interface Props {
+  showDecimal: boolean
+  onDigitPress: (digit: number) => void
+  onBackspacePress: () => void
+  onDecimalPress?: () => void
+}
+
+function DigitButton({
+  digit,
+  onDigitPress,
+}: {
+  digit: number
+  onDigitPress: (digit: number) => void
+}) {
+  const onPress = () => onDigitPress(digit)
+  return (
+    <Touchable borderless={true} onPress={onPress}>
+      <Text style={style.digit}>{digit}</Text>
+    </Touchable>
+  )
+}
+
+export default function NumberKeypad(props: Props) {
+  return (
+    <View style={style.container}>
+      <View style={style.row}>
+        <DigitButton digit={1} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={2} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={3} onDigitPress={props.onDigitPress} />
+      </View>
+      <View style={style.row}>
+        <DigitButton digit={4} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={5} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={6} onDigitPress={props.onDigitPress} />
+      </View>
+      <View style={style.row}>
+        <DigitButton digit={7} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={8} onDigitPress={props.onDigitPress} />
+        <DigitButton digit={9} onDigitPress={props.onDigitPress} />
+      </View>
+      <View style={style.row}>
+        {props.showDecimal ? (
+          <Touchable borderless={true} onPress={props.onDecimalPress}>
+            <Text style={style.digit}>.</Text>
```

Comment:
This will be incorrect for some locales.

Reply:
Yep, I haven't i18n'ed this yet, will add a TODO